### PR TITLE
[feature] Added phone number to register and login #144

### LINKED ISSUE #144: [change] Use mobile phone to register and log in when mobile phone verification is enabled
When mobile verification is enabled for an org, implement the following changes:

- right now, the [email is used as the username when registering to openwisp-radius](https://github.com/openwisp/openwisp-wifi-login-pages/blob/master/client/components/registration/registration.js#L62); when mobile phone verification is enabled, the phone number should be used instead
- in the login component, when mobile phone verification is enabled, the [phone number input field](https://github.com/openwisp/openwisp-wifi-login-pages/blob/master/client/components/registration/registration.js#L165-L182) should be shown instead of the username, with the same options that are passed used during registration

### PR DESCRIPTION
**Pull request by NoumbissiValere**
Closes #144

### CODE REVIEW

**Comment by NoumbissiValere** [comment]
This PR needs [openwisp_radius #209](https://github.com/openwisp/openwisp-radius/pull/210) for testing

File: client/components/login/login.js [hunk 1]
```diff
@@ -174,31 +179,66 @@ export default class Login extends React.Component {
...
-                    <div className="error">
-                      <span className="icon">!</span>
-                      <span className="text">
-                        {errors.email}
-                      </span>
-                    </div>
-                  )}
-                  <input
-                    className={`input ${errors.email ? "error" : ""}`}
-                    type={input_fields.email.type}
```
Comment (nemesisdesign) [inline client/components/login/login.js:191]: please ensure the configuration mirrors and reuses the same configuration used for the registration component! Do as it's done in the `mobile-phone-change` component, if you inspect that component (check `index.js`) you'll see the configuration is taken from the registration component:

https://github.com/openwisp/openwisp-wifi-login-pages/blob/f4fee4b1ab52156d3026296120ba92c1d1c2f1d5/client/components/mobile-phone-change/index.js#L9-L11

Ensure also the HTML is exactly the same as in the change phone number and registration component.

File: client/components/login/login.js [hunk 2]
```diff
@@ -174,31 +179,66 @@ export default class Login extends React.Component {
...
+                      name="email"
+                      value={email}
+                      onChange={this.handleChange}
+                      required
+                      placeholder={getText(input_fields.email.placeholder, language)}
+                      pattern={input_fields.email.pattern}
+                      title={getText(input_fields.email.pattern_description, language)}
+                    />
+                  </div>
+                )}
```
Comment (nemesisdesign) [inline client/components/login/login.js:241]: I think we could make a method here:
```javascript
{this.renderLoginField()}
```

```javascript
renderLoginField() {
    if (settings.mobile_phone_verification) {
        return this.renderPhoneField();
    }
    return this.renderEmailField();
}
```

See also https://reactjs.org/docs/conditional-rendering.html.

File: internals/generators/organization/config.yml.hbs [hunk 3]
```diff
@@ -354,6 +354,11 @@ client:
             en: "password"
           label:
             en: "password"
+        phone_number:
+          placeholder:
+            en: "enter mobile phone number"
+          label:
+            en: "mobile phone number"
```
Comment (nemesisdesign) [inline internals/generators/organization/config.yml.hbs:361]: if you follow the suggestions above, this should not be needed

File: org-configurations/default-configuration.yml [hunk 4]
```diff
@@ -267,6 +267,11 @@ client:
             en: "password"
           label:
             en: "password"
+        phone_number:
+          placeholder:
+            en: "enter mobile phone number"
+          label:
+            en: "mobile phone number"
```
Comment (nemesisdesign) [inline org-configurations/default-configuration.yml:274]: if you follow the suggestions above, this should not be needed

File: client/components/login/login.test.js [hunk 5]
```diff
@@ -219,10 +222,12 @@ describe("<Login /> interactions", () => {
...
 
+    expect(wrapper.find("#email")).toEqual({});
+    expect(wrapper.state("phone_number")).toEqual("");
     wrapper
-      .find("#email")
-      .simulate("change", { target: { value: "test email", name: "email" } });
-    expect(wrapper.state("email")).toEqual("test email");
+      .find("[name='phone_number']")
+      .simulate("change", {target: {value: "+393660011333", name: "phone_number"}});
+    expect(wrapper.state("phone_number")).not.toEqual("");
```
Comment (nemesisdesign) [inline client/components/login/login.test.js:230]: ensure there's 1 test for the email scenario and 1 for the phone number scenario please

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Good progress, see below.

File: client/test-config.json [hunk 6]
```diff
@@ -372,4 +380,4 @@
       }
     }
   }
-]
+]
```
Comment (nemesisdesign) [inline client/test-config.json:383]: final blank line missing here

File: org-configurations/default-configuration.yml [hunk 7]
```diff
@@ -77,6 +77,15 @@ client:
...
       input_fields:
+        phone_number:
+          placeholder:
+            en: "enter mobile phone number"
+          label:
+            en: "mobile phone number"
+          # default international prefix
+          # country: "it"
+          # restrict international prefixes shown
+          # only_countries: ["it", "de", "uk"]
```
Comment (nemesisdesign) [inline org-configurations/default-configuration.yml:88]: remove the commented options please

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**